core/services/job/orm.go

```diff
@@ -214,14 +214,23 @@ func (o *orm) unclaimJob(ctx context.Context, id int32) error {
 }
 
 func (o *orm) RecordError(ctx context.Context, jobID int32, description string) {
+	// Noop if the job has been deleted.
+	err := o.db.First(&models.JobSpecV2{}, "id = ?", jobID).Error
+	if err == gorm.ErrRecordNotFound {
+		return
+	} else if err != nil {
+		logger.Errorf("error checking for job existence %v", err)
+		return
+	}
```

Comment:
Why bother checking this here, instead of checking for foreign key constraint on the insert?

Reply:
just because then you need to do a fuzzy match on some substring in the error which seems brittle

Reply:
We do this in other areas in the code, e.g: `ErrOptimisticUpdateConflict` and although it feels brittle, most of those error messages have been fairly consistent for 20+ years, and the code is more performant, simpler and avoids any race conditions!

Reply:
fair enough - updated!